src/corehost/corehost.cpp

```diff
@@ -307,7 +325,7 @@ int get_com_activation_delegate(pal::string_t *app_path, com_activation_fn *dele
     app_path_local.replace(app_path_local.begin() + idx, app_path_local.end(), _X(".dll"));
 
     *app_path = std::move(app_path_local);
-    return entry_del(host_path.c_str(), dotnet_root.c_str(), app_path->c_str(), (void**)delegate);
+    return get_com_delegate(host_path.c_str(), dotnet_root.c_str(), app_path->c_str(), (void**)delegate);
```

Comment:
You should reset the error writer here to its original value. To make things clean.
There's a `propagate_error_writer` class in utils to make this simple and robust.